From 0f6c8431f1ecebb019d7263d285e5f6e34cb87bb Mon Sep 17 00:00:00 2001
From: Viorel Suman <viorel.suman@nxp.com>
Date: Mon, 27 Aug 2018 14:38:24 +0300
Subject: [PATCH 4454/5242] MLK-18898-4: ARM64: dts: imx8mq-evk: add SAI1
 pcm_b2m pinctrl state

commit  8a4873e022c0f6aec7b83bb84d4b7bf21216836d from
https://source.codeaurora.org/external/imx/linux-imx.git

Add SAI1 pcm_b2m pinctrl state needed to map SAI1 BCLK to codec
MCLK pin.

Signed-off-by: Viorel Suman <viorel.suman@nxp.com>
(cherry picked from commit 63c362c8c6cbe1e3a6b752ada828418e3c3e4363)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts |   24 +++++++++++++++++++---
 1 file changed, 21 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
index 78df263..b4bfc3b 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
@@ -354,7 +354,7 @@
 			>;
 		};
 
-		pinctrl_sai1: sai1grp {
+		pinctrl_sai1_pcm: sai1grp_pcm {
 			fsl,pins = <
 				MX8MQ_IOMUXC_SAI1_MCLK_SAI1_MCLK		0xd6
 				MX8MQ_IOMUXC_SAI1_TXFS_SAI1_TX_SYNC		0xd6
@@ -371,6 +371,23 @@
 			>;
 		};
 
+		pinctrl_sai1_pcm_b2m: sai1grp_pcm_b2m {
+			fsl,pins = <
+				MX8MQ_IOMUXC_SAI1_MCLK_SAI1_TX_BCLK		0xd6
+				MX8MQ_IOMUXC_SAI1_TXFS_SAI1_TX_SYNC		0xd6
+				MX8MQ_IOMUXC_SAI1_RXD7_SAI1_TX_SYNC		0xd6
+				MX8MQ_IOMUXC_SAI1_TXC_SAI1_TX_BCLK		0xd6
+				MX8MQ_IOMUXC_SAI1_TXD0_SAI1_TX_DATA0		0xd6
+				MX8MQ_IOMUXC_SAI1_TXD1_SAI1_TX_DATA1		0xd6
+				MX8MQ_IOMUXC_SAI1_TXD2_SAI1_TX_DATA2		0xd6
+				MX8MQ_IOMUXC_SAI1_TXD3_SAI1_TX_DATA3		0xd6
+				MX8MQ_IOMUXC_SAI1_TXD4_SAI1_TX_DATA4		0xd6
+				MX8MQ_IOMUXC_SAI1_TXD5_SAI1_TX_DATA5		0xd6
+				MX8MQ_IOMUXC_SAI1_TXD6_SAI1_TX_DATA6		0xd6
+				MX8MQ_IOMUXC_SAI1_TXD7_SAI1_TX_DATA7		0xd6
+			>;
+		};
+
 		pinctrl_sai2: sai2grp {
 			fsl,pins = <
 				MX8MQ_IOMUXC_SAI2_TXFS_SAI2_TX_SYNC	0xd6
@@ -759,8 +776,9 @@
 };
 
 &sai1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_sai1>;
+	pinctrl-names = "default", "pcm_b2m";
+	pinctrl-0 = <&pinctrl_sai1_pcm>;
+	pinctrl-1 = <&pinctrl_sai1_pcm_b2m>;
 	assigned-clocks = <&clk IMX8MQ_CLK_SAI1_SRC>,
 			<&clk IMX8MQ_CLK_SAI1_DIV>;
 	assigned-clock-parents = <&clk IMX8MQ_AUDIO_PLL1_OUT>;
-- 
1.7.9.5

